http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -179,6 +179,91 @@ class HttpSamDAO[F[_]: Effect](httpClient: Client[F], config: HttpSamDaoConfig,
       }
     } yield ()
 
+  //Notifications that Leo has created/destroyed persistent disks. Allows the auth provider to register things.
+  /**
+    * Leo calls this method to notify the auth provider that a new persistent disk has been created.
+    * The returned future should complete once the provider has finished doing any associated work.
+    * Returning a failed Future will prevent the disk from being created, and will call notifyPersistentDiskDeleted for the same disk.
+    * Leo will wait, so be timely!
+    *
+    * @param internalId     The internal ID for the disk (i.e. used for Sam resources)
+    * @param creatorEmail   The email address of the user in question
+    * @param googleProject  The Google project the disk was created in
+    * @return A Future that will complete when the auth provider has finished doing its business.
+    */
+  def createPersistentDiskResource(internalId: PersistentDiskInternalId,
```

Comment:
Is the only difference between `create/deletePersistentDiskResource` and `create/deleteClusterResource` the Sam URL? Maybe some code could be shared.

Reply:
If as part of your PD data model PR you're refactoring some types out to be used both by PDs and runtimes, we can add a `SamResourceId` trait there, and then `PersistentDiskSamResourceId` and `RuntimeSamResourceId` can both extend from that - I think that would help in making this code more easily sharable between resource types

Reply:
Yeah  good point -- I think we can merge this PR as is and I can rebase and refactor the models.